### PR TITLE
add bar door and trapdoor

### PR DESCRIPTION
**Pull request by TumeniNodes**
had to redo the PR altogether as I made a mistake in rebasing and something got hung up.
This new PR replaces https://github.com/minetest/minetest_game/pull/2443
It also registers the bar doors within 'xpanes' as requested by both paramat & sfan5
screenshot_20190920_121814

### CODE REVIEW

File: mods/xpanes/init.lua [hunk 1]
```diff
@@ -202,3 +202,41 @@ minetest.register_lbm({
 		end
 	end
 })
+
+-- Register steel bar doors and trapdoors
+
+doors.register("xpanes:door_steel_bar", {
```
Comment (sfan5) [inline mods/xpanes/init.lua]: since doors is an optional dependency, you have to check if it was *actually* loaded here

Comment (TumeniNodes) [reply mods/xpanes/init.lua]: check added

**Comment by TumeniNodes** [comment]
Thank you

File: mods/xpanes/init.lua [hunk 2]
```diff
@@ -202,3 +202,45 @@ minetest.register_lbm({
 		end
 	end
 })
+
+-- Register steel bar doors and trapdoors
+
+if minetest.get_modpath ("doors") then
```
Comment (paramat) [inline mods/xpanes/init.lua]: No space after
`minetest.get_modpath`

File: mods/xpanes/init.lua [hunk 3]
```diff
@@ -202,3 +202,45 @@ minetest.register_lbm({
...
+	sound_close = "doors_steel_door_close",
+})
+
+minetest.register_craft({
+	output = "xpanes:trapdoor_steel_bar",
+	recipe = {
+		{"xpanes:bar_flat", "xpanes:bar_flat"},
+		{"xpanes:bar_flat", "xpanes:bar_flat"},
+	}
+})
```
Comment (paramat) [inline mods/xpanes/init.lua]: The code block above needs indenting.

**Comment by paramat** [comment]
No objections from me.

File: mods/xpanes/init.lua (hunk 2 above)
Comment (TumeniNodes) [reply mods/xpanes/init.lua]: fixed

File: mods/xpanes/init.lua (hunk 3 above)
Comment (TumeniNodes) [reply mods/xpanes/init.lua]: and fixed.

**Comment by TumeniNodes** [comment]
> No objections from me.

Hey, that sounds like two approvals :P

**Comment by paramat** [comment]
Nuuu it isn't, it means no objections but no approval =) I'm neutral.

**Comment by TumeniNodes** [comment]
thank you

File: mods/xpanes/license.txt [hunk 4]
```diff
@@ -34,6 +34,7 @@ Attribution-ShareAlike 3.0 Unported (CC BY-SA 3.0)
 Copyright (C) 2014-2016 xyz
 Copyright (C) 2013-2016 Gambit
 Copyright (C) 2016 paramat
+copyright (C) 2016-2019 TumeniNodes
```
Comment (paramat) [inline mods/xpanes/license.txt]: Copyright

**Comment by paramat** [comment]
Texture error:
xpanes_trapdoor_steel_bar_side is still RGB

File: mods/xpanes/license.txt (hunk 4 above)
Comment (paramat) [reply mods/xpanes/license.txt]: Also, why from 2016? Aren't the textures made recently?

Comment (TumeniNodes) [reply mods/xpanes/license.txt]: yes, corrected.

**Comment by TumeniNodes** [comment]
> Texture error:
> 'xpanes_trapdoor_steel_bar_side' is still RGB
> I'll merge this once those 2 things are fixed.

Both have been corrected.

**Pull request merged**